src/python_testing/TC_SEAR_1_2.py

```diff
@@ -0,0 +1,384 @@
+            if a.areaID in areaid_list:
+                asserts.fail("SupportedAreas must have unique AreaID values!")
+            else:
+                areaid_list.append(a.areaID)
+                
+                if len(self.mapid_list) > 0:
+                    asserts.assert_is_not(a.mapID, NullValue,
+                             f"SupportedAreas entry with AreaID({a.areaID}) should not have null MapID")
+                    asserts.assert_is(a.mapID in self.mapid_list,
+                             f"SupportedAreas entry with AreaID({a.areaID}) has unknown MapID({a.mapID})")
+                    areadesc_same_map_list = []
+                    areadesc_same_map_list.append(a.areaDesc)
+                    for b in supported_areas:
+                        if a.mapID == b.mapID:
+                            if b.areaDesc in areadesc_same_map_list:
+                                asserts.fail(f"SupportedAreas must have unique MapID({a.mapID}) + AreaDesc({a.areaDesc}) values!")
+                            else:
+                                areadesc_same_map_list.append(b.areaDesc)
+                else:
+                    #empty SupportedMaps
+                    asserts.assert_is(a.mapID, NullValue,
+                             f"SupportedAreas entry with AreaID({a.areaID}) should have null MapID")
+                    areadesc_list = []
+                    areadesc_list.append(a.areaDesc)
+                    for b in supported_areas:
+                        if b.areaDesc in areadesc_list:
+                                asserts.fail(f"SupportedAreas must have unique AreaDesc({a.areaDesc}) values!")
+                        else:
+                            areadesc_list.append(b.areaDesc)
+                if a.locationInfo is NullValue and a.landmarkTag is NullValue:
+                    asserts.assert_true(f"SupportedAreas entry with AreaID({a.areaID}) should not have null LocationInfo and null LandmarkTag")
+                if a.landmarkTag is not NullValue:
+                    asserts.assert_true(a.landmarkTag in range(0, self.MAX_LANDMARK_ID),
+                             f"SupportedAreas entry with AreaID({a.areaID}) has invalid LandmarkTag({a.landmarkTag})")
+                    asserts.assert_true(a.positionTag is NullValue or a.positionTag in range(0, self.MAX_RELPOS_ID),
+                             f"SupportedAreas entry with AreaID({a.areaID}) has invalid PositionTag({a.positionTag})")
+        #save so other methods can use this if neeeded
+        self.areaid_list = areaid_list
+
+    async def read_and_validate_selected_areas(self, step):
+        self.print_step(step, "Read SelectedAreas attribute")
+        selected_areas = await self.read_sear_attribute_expect_success(
+            endpoint=self.endpoint, attribute=Clusters.ServiceArea.Attributes.SelectedAreas)
+        logging.info(f"SelectedAreas {selected_areas}")
+
+        #TODO how to check if all entries are uint32?
```

Comment:
Can you check if the values are >= 0 and < 2^32?

Reply:
I wonder if we need to do this type of type checks in general. Perhaps the python Matter test infra is already doing that.